pulsar-common/src/main/java/com/yahoo/pulsar/common/api/Commands.java

```diff
@@ -416,11 +416,15 @@ public static ByteBuf newLookup(String topic, boolean authoritative, long reques
     public static ByteBuf newLookupResponse(String brokerServiceUrl, String brokerServiceUrlTls, boolean authoritative,
             LookupType response, long requestId) {
         CommandLookupTopicResponse.Builder connectionBuilder = CommandLookupTopicResponse.newBuilder();
-        connectionBuilder.setBrokerServiceUrl(brokerServiceUrl);
+        if (brokerServiceUrl != null) {
+            connectionBuilder.setBrokerServiceUrl(brokerServiceUrl);
```

Comment:
If we don't have service url, shouldn't we respond with some sort of error response?